R/human_infection.R

```diff
@@ -519,17 +519,61 @@ relapse_bite_infection_hazard_resolution <- function(
   ## all bitten humans with an infectious bite (incorporating prophylaxis) get a new batch of hypnozoites
   if(bite_infections$size()>0){
     
+    ls_prophylaxis <- ls_treatment_prophylaxis_efficacy(
+      bite_infections,
+      variables,
+      parameters,
+      timestep
+    )
+    
     # make sure batches are capped
-    current_batches <- variables$hypnozoites$get_values(bite_infections)
+    new_hypnozoite_batch_formed <- bitset_at(bite_infections, bernoulli_multi_p(1 - ls_prophylaxis))
+    current_batches <- variables$hypnozoites$get_values(new_hypnozoite_batch_formed)
     new_batch_number <- pmin(current_batches + 1, parameters$kmax)
     
     variables$hypnozoites$queue_update(
       new_batch_number,
-      bite_infections
+      new_hypnozoite_batch_formed
     )
   }
 }
 
+#' @title Calculate protection from formation of new hypnozoite batches due to liver stage drug prophylaxis
+#' @description This function calculates the probability that liver stage drug prophylaxis will 
+#' protect each individual from the formation of new hynozoite batches
+#' @param bite_infections a vector of individuals with prospective formation of new hypnozoite batches
+#' @param variables a list of all of the model variables
+#' @param parameters model parameters
+#' @param timestep current timestep
+#' @noRd
+ls_treatment_prophylaxis_efficacy <- function(
+    bite_infections,
+    variables,
+    parameters,
+    timestep
+){
+  
+  ## drug prophylaxis may limit formation of new hypnozoite batches
+  ls_prophylaxis <- rep(0, bite_infections$size())
+  if(length(parameters$drug_hypnozoite_efficacy) > 0){
+
+    ls_drug <- variables$ls_drug$get_values(bite_infections)
+    ls_medicated <- ls_drug > 0
+    ls_medicated[ls_medicated] <- !is.na(parameters$drug_hypnozoite_efficacy[ls_drug])
```

Comment:
This is quite confusing. And possibly redundant? Don't we expect all liver stage drugs have a hypnozoite efficacy?

Reply:
If I understand the question, the way I've coded it is that the parameter sets are parasite specific, rather than being blood or liver stage specific. That means that vivax drug parameter sets can be either only blood stage (where they would have 4 parameters), or would have both blood and liver stage parameters (which total 7 parameters), and not all "vivax drugs" have liver stage action. Does that answer the question? (Happy to change things if you have a better way of doing things!)

Reply:
Would it be more consistent to have liver-stage parameters for each vivax drug, and just set them to 0 if the drug is assumed to have no liver-stage impact?